test/integration/backward_compatible/parallel/topic/ReliableTopicTest.js

```diff
@@ -93,6 +99,20 @@ describe('ReliableTopicTest', function () {
         }).catch(done);
     });
 
+    it('writes and reads messages with addListener method', async function () {
+        const topicName = 't' + Math.random();
+        const topicOne = await clientOne.getReliableTopic(topicName);
+        const topicTwo = await clientTwo.getReliableTopic(topicName);
+        await topicTwo.addListener(async (msg) => {
+            if (msg.messageObject['value'] !== 'foo') {
+                throw new Error('Message received does not match expected value.');
+            }
+        });
```

Comment:
This test does not wait for the listener to run. It is not a valid test. You can create a deferred promise and await it in the test while resolving it in the addListener callback. Alternatively, you can use done syntax(which I don't recommend)